src/network/protocol_handler.cpp

```diff
@@ -359,19 +361,20 @@ void ProtocolHandler::ExecQueryMessage(InputPacket *pkt, const size_t thread_id)
     std::string error_message;
     int rows_affected = 0;
 
-    std::string query_type_string_;
-    Statement::ParseQueryTypeString(query, query_type_string_);
-
+    std::string query_type_string;
+    Statement::ParseQueryTypeString(query, query_type_string);
     QueryType query_type;
-    Statement::MapToQueryType(query_type_string_, query_type);
-    std::stringstream stream(query_type_string_);
+    Statement::MapToQueryType(query_type_string, query_type);
 
     switch (query_type) {
       case QueryType::QUERY_PREPARE:
       {
         std::string statement_name;
-        stream >> statement_name;
-        std::size_t pos = query.find("AS");
+        std::vector<std::string> tokens;
+        boost::split(tokens, query, boost::is_any_of("(), "));
```

Comment:
This seems like a hack. Why do we want to do this? Why can't this be done in the parser?

Reply:
Here it is to get the real statement out of a PREPARE query, such as 'prepare f as insert into foo values($1)'. It will get the statement as 'insert into foo values($1)', and store it in the statement cache with name 'f'. Then it will put the 'f' statement into the parser and build a plan tree. Right now it is done in the protocol handler rather than parser. The parser won't take care of the PREPARE query.
The use of 'split' seems hack. I can go to see how postgres handles PREPARE query.
Can I ask that, do you mean this should be done in the parser instead of protocol handler, or the use of 'split' is too hack?

Reply:
By testing the parser, it seems parser doesn't parse the PREPARE query with 'prepare' tag. Let me see how it is working and make sure

Reply:
Coordinate with @haojin2 on this. He was in charge of the new parser last semester.

Reply:
Parser can handle this. Sorry I made a mistake when testing. 
Right now in protocol_handler, there are several places where we use functions like split(), find() to analyze the query, before putting it into the parser, like the way to get query_type and handle PREPARE query and EXECUTE query. Shall we change it now?

Reply:
Yes change it now.